src/main/kotlin/calculator/StringAddCalculator.kt

```diff
@@ -0,0 +1,23 @@
+package calculator
+
+import java.lang.RuntimeException
+
+class StringAddCalculator (
+    private val spliterator: Spliterator<String>,
+    private val validation: NumberValidation
+) {
+
+    fun add(input: String?): Int {
+        require(input != null && input != "") { return 0 }
```

Comment:
require문은 기본적으로 block문안의 문자열 메세지를 IllegalArgumentException으로 던지기 위한 검증 함수입니다. 
강제로 return 0 하는 것 보다는 다음과 같이 if문으로 해도 될 것 같네요!
```suggestion
        if(isNullOrBlank()) { 
            return 0 
        }
```